### ISSUE TITLE
Q: Package to AppImage

### ISSUE THREAD

**Issue by territory-sschneider** [srevinsaju/zap#11]
Is it possible to get a package and convert it into an AppImage on the fly? seems like on appimagehub some stuff gets outdated pretty quickly.

**Comment by srevinsaju** [comment]
Yes, its the `pkg2appimage` repo: see https://github.com/AppImage/pkg2appimage

**Comment by territory-sschneider** [comment]
does it only work for deb, ppa or also for AUR?

**Comment by srevinsaju** [comment]
Yes. Which app are you looking for?

**Comment by territory-sschneider** [comment]
Well just in general to avoid some system update issues. Lately, I ran into an issue with python for example which rendered khal unusable because of some updated package.
________________________________

**Comment by territory-sschneider** [comment]
But in general it would be useful just to be able to make a appimage out of some stuff to have the system a bit more tidy. On the otherhand, having to have to make the recepies for every package would be tedious also

**Comment by srevinsaju** [comment]
It depends on the package. I have many appimage recipes on my GitHub user, which provides up to date AppImages. If you would like to have an up-to-date appimage, I might look into it. I would create appimages, if its useful for both me and you.

**Comment by territory-sschneider** [comment]
I made a short lit :)
alacritty 0.6.0-1
cawbird 1.2.1-1
chromium 87.0.4280.88-2
ctop 0.7.5-1
dbeaver 7.3.0-1
docker 1:20.10.1-1
docker-compose 1.27.4-2
ferdi 5.5.0-8
flameshot 0.8.5-1
kakoune 2020.09.01-1
khal 0.10.2-2
kitty 0.19.3-1
meld 3.20.2-2
nitrogen 1.6.1-3
p4v 2020.3.2060285-1
parcellite 1.2.1-4
picom 8.2-1
qtile 0.16.1-2
qutebrowser 1.14.1-1
ranger 1.9.3.143.g5047d9e5-1
realvnc-vnc-viewer 6.20.529-1
rofi 1.6.1-1
speedcrunch 0.12.0-3
stacer 1.1.0-1.2
starship 0.48.0-1

**Comment by srevinsaju** [comment]
Iirc, `flameshot` has an appimage on their official repository. You might like to go there. I will check the rest.


PS: I can't imagine having a `docker` appimage 😂

**Comment by territory-sschneider** [comment]
that's why i put it in there :)

**Comment by srevinsaju** [comment]
Instead of `chromium`, I would suggest you to take a look at the [Brave AppImage](https://github.com/srevinsaju/Brave-AppImage). Under the hood, its the chromium engine.

**Comment by territory-sschneider** [comment]
Can you extend zap to auto generate appimages if none is available? or generate appimages from git repos or something like that?

**Comment by srevinsaju** [comment]
I wish I could. Creating AppImage are not an easy task however. It takes a lot of time, trial and error. So the recipes need to be created from scratch. However, I can extend zap to include all the apps which exist in `pkg2appimage` so that, a new appimage would be built from the recipe provided at `pkg2appimage`.

**Comment by territory-sschneider** [comment]
teams-for-linux-official-bin

**Comment by territory-sschneider** [comment]
Maybe you can explain this too me.
I use qtile and have the prtscr key bound to `flameshot gui` it works with the AUR but it doesn't work with appimage. Why?

**Comment by srevinsaju** [comment]
Same, the Flameshot AppImage doesn't get triggered here too. And I don't know why :crying_cat_face:

**Comment by territory-sschneider** [comment]
What runs the appImage? maybe it is the way the appImage is called like not being marked as executable or such.

**Comment by srevinsaju** [comment]
There is _nothing_ that runs the appimage. It runs on its own. Thats the AppImage philosophy.

**Comment by territory-sschneider** [comment]
oh well, then i'm out of ideas

**Comment by srevinsaju** [comment]
We will have to ask @flameshot-org about it. They should know.

**Comment by srevinsaju** [comment]
Ask them here: https://github.com/flameshot-org/flameshot/issues

**Comment by territory-sschneider** [comment]
Is there a way to integrate a downloaded appimage to zap? like, from file.

**Comment by srevinsaju** [comment]
```bash
zap install <nameofappimage>  --from file:///path/to/file.AppImage
```

**Comment by territory-sschneider** [comment]
hey, you could add zap to https://github.com/tldr-pages/tldr :)
